tests/test_harness.rs

```diff
@@ -0,0 +1,75 @@
+use std::{
+    env::{self, join_paths},
+    fs::canonicalize,
+    io::{Error, ErrorKind},
+    process::{Command, ExitStatus},
+};
+
+pub struct Harness;
+
+type Result<T = ()> = std::io::Result<T>;
+
+/// Maps an exit status to a result, meaning that a non-successful exit status
+/// will get mapped to an `Err` variant
+trait ExitStatusConverter {
+    fn into_result(self) -> Result;
+}
+
+impl ExitStatusConverter for Result<ExitStatus> {
+    fn into_result(self) -> Result {
+        match self {
+            Ok(exit_code) => match exit_code.success() {
+                true => Ok(()),
+                false => Err(Error::new(
+                    ErrorKind::Other,
+                    "command did not exit successfully",
+                )),
+            },
+            Err(e) => Err(e),
+        }
+    }
+}
+
+impl Harness {
+    /// Build the project present in the current directory using `rustc` or `gccrs`
+    fn cargo_build(use_gccrs: bool) -> Result {
+        let mut cmd = Command::new("cargo");
+
+        if use_gccrs {
+            // Add `gccrs` argument so that `cargo build` becomes `cargo gccrs build`
+            cmd.arg("gccrs");
+
+            // Tweak the path so that the most recently compiled *debug* version of
+            // cargo-gccrs is available as a subcommand
+            let mut paths = vec![canonicalize("../../target/debug")?];
+            paths.append(&mut env::split_paths(&env::var("PATH").unwrap()).collect::<Vec<_>>());
+
+            let new_path = join_paths(paths.iter()).unwrap();
+
+            cmd.env("PATH", new_path);
+        }
+
+        cmd.arg("build").status().into_result()
```

Comment:
Why not just call 
```suggestion
        cmd.arg("build").status().and_then(|s| if s.success() { Ok(()) } else { Err(Error::new(...)) } )

Reply:
You're right, I thought I'd reuse this code in more places. Fixed in 978d1bd